clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -74,7 +76,13 @@ private Utils() {}
     private static final Pattern VALID_HOST_CHARACTERS = Pattern.compile("([0-9a-zA-Z\\-%._:]*)");
 
     // Prints up to 2 decimal digits. Used for human readable printing
-    private static final DecimalFormat TWO_DIGIT_FORMAT = new DecimalFormat("0.##");
+    private static final DecimalFormat TWO_DIGIT_FORMAT;
```

Comment:
I think we can just use `NumberFormat` and avoid the cast below.

Reply:
@hachikuji If I don't cast to DecimalFormat then I can't do .applyPattern("0.##")

If I use NumberFormat then I have to do the following in the static initializer to make it work:
        TWO_DIGIT_FORMAT = NumberFormat.getNumberInstance(Locale.ENGLISH);
        TWO_DIGIT_FORMAT.setGroupingUsed(false);
        TWO_DIGIT_FORMAT.setMaximumFractionDigits(2);

Both ways work, but as even the documentation suggests the cast it might be a bit more readable.
WDYT?

Reply:
Hmm, I see.. What an annoying API. I guess another way is something like this:
```java
private static final DecimalFormat TWO_DIGIT_FORMAT = new DecimalFormat("0.##", 
  DecimalFormatSymbols.getInstance(Locale.ENGLISH));
```
Does that work?

Reply:
Yes, that works too.
Even the JavaDoc for this constructor says:
     * To obtain standard formats for a given
     * locale, use the factory methods on NumberFormat such as
     * getInstance or getCurrencyInstance. If you need only minor adjustments
     * to a standard format, you can modify the format returned by
     * a NumberFormat factory method.

As interesting as this might be for a learning experience, I would rather get over it ;-)
Pushed your suggestion, hope @dongjinleekr does not object.